### PR TITLE
Notebook save/open dialog boxes

### LINKED ISSUE #247: Improve save/load behavior
**Save as file dialog**
- Show full file path below input box (see mockups)
  - If user has input a relative path, expand PWD
- Append .bkr if user doesn’t include it
- If user selects a directory, show dialog with “{selected path} is a directory. Please choose a different location” Include a close button that takes user back to ‘save as dialog’ with the same state as right before the clicked save.
- If the file exists, show dialog with “{selected path} already exists. Overwrite?” with buttons ‘Overwrite’ and ‘Cancel’ Cancel should take user back to ‘save as dialog’

**Open file dialog**
- Checkbox to show only .bkr files, checked by detault (see mockup)

**Save errors**
- Show dialog with title ‘Save failed’, text save failed + short exception, scroll pane with full exception, and a ‘close’ button.

**Open errors**
- For errors accessing file, use the same type of dialog as save errors
- If the content isn’t a notebook, show a dialog to that effect (currently this is done with alert(), let’s use a Bootstrap dialog for consistency)

save-as-file-pwd

save-as-file-pwd-2

### PR DESCRIPTION
**Pull request by anglee**
Fix #247

### CODE REVIEW

File: core/src/main/web/app/mainapp/mainapp.js [hunk 1]
```diff
@@ -206,51 +206,142 @@
         };
 
         var _impl = (function() {
-          var _saveNotebook = function() {
-            showStatusMessage("Saving");
+
+          var promptUriChooser = function(initUri) {
```
Comment (anglee) [inline core/src/main/web/app/mainapp/mainapp.js:214]: see https://www.lucidchart.com/documents/view/d2c35b6f-cbb6-48bd-be0d-80c86bb97a04

**Comment by jchendy** [comment]
Help and settings menus seem to be missing from control panel.

**Comment by jchendy** [comment]
Also 'open -> .bkr' is missing.

**Comment by jchendy** [comment]
I found out I had the same problem on master. Ang suggested deleting beaker.conf.json, which worked.

**Comment by jchendy** [comment]
As per the spec, the buttons on the 'file exists' dialog should say 'Overwrite' and 'Cancel' instead of yes/no. Please also use Bootstrap button styles with btn-danger for 'Overwrite' and the default class for 'Cancel'

**Comment by jchendy** [comment]
This may have been an omission in the spec, but names like '~/file' don't expand properly in the save dialog. Can we have it translate into '/User/jeff/file.bkr' etc

**Comment by jchendy** [comment]
Two issues related to opening .ipynb:
- Open an IPython file and click save. It wipes out the file with a Beaker file. We had previously decided that we don't want to overwrite converted files, so this should bring up the 'save as' dialog
- Open an IPython file, save it somewhere else, and load that file. I get 'TypeError: Cannot read property '0' of undefined'

**Comment by jchendy** [comment]
Also, for the overwrite dialog, can you please make 'overwrite' the right button and add the class btn-danger?

**Comment by jchendy** [comment]
Please swap the order of 'overwrite' and 'cancel' buttons. We should always have the 'action' or 'go forward' button on the right.

**Comment by jchendy** [comment]
1) Open a notebook
2) Try to open an invalid file
3) After you close the dialog, you end up with a corrupted version of the original notebook that can't run any code

Let's open the control panel instead.

**Comment by jchendy** [comment]
On the 'save failed' dialog, can we please have the blue 'close' button in the bottom right for consistency with our other dialogs?

**Comment by jchendy** [comment]
Sorry, found another thing. The 'show .bkr files only' label should be clickable and toggle the checkbox.

**Pull request merged**